backend/app/services/observability/PostgresClient.scala

```diff
@@ -150,23 +157,24 @@ class PostgresClientImpl(postgresConfig: PostgresConfig) extends PostgresClient
             -- You can't array_agg arrays of varying cardinality so here we convert to string
             ARRAY_REMOVE(ARRAY_AGG(ARRAY_TO_STRING(extractor_statuses.extractor_event_times, ',','null')), NULL) AS "extractorEventTimes",
             ARRAY_REMOVE(ARRAY_AGG(ARRAY_TO_STRING(extractor_statuses.extractor_event_statuses, ',','null')), NULL) AS "extractorStatuses"
-            FROM (
-              SELECT
-                blob_id,
-                ingest_id,
-                MIN(EXTRACT(EPOCH from event_time)) AS ingest_start,
-                MAX(EXTRACT(EPOCH from event_time)) AS most_recent_event,
-                ARRAY_AGG(details -> 'errors') as errors,
-                (ARRAY_AGG(details ->> 'workspaceName')  FILTER (WHERE details ->> 'workspaceName' IS NOT NULL))[1] as workspace_name,
-                (ARRAY_AGG(details ->> 'mimeTypes')  FILTER (WHERE details ->> 'mimeTypes' IS NOT NULL))[1] as mime_types
-              FROM ingestion_events
-              WHERE ingest_id LIKE ${if(ingestIdIsPrefix) LikeConditionEscapeUtil.beginsWith(ingestId) else ingestId}
-              GROUP BY 1,2
-            ) AS ie
-            LEFT JOIN blob_metadata USING(ingest_id, blob_id)
-            LEFT JOIN extractor_statuses on extractor_statuses.blob_id = ie.blob_id and extractor_statuses.ingest_id = ie.ingest_id
-            GROUP BY 1,2,3,4,5,6,7
-            ORDER by ingest_start desc
+          FROM (
+            SELECT
+              blob_id,
+              ingest_id,
+              MIN(EXTRACT(EPOCH from event_time)) AS ingest_start,
+              MAX(EXTRACT(EPOCH from event_time)) AS most_recent_event,
+              ARRAY_AGG(details -> 'errors') as errors,
+              (ARRAY_AGG(details ->> 'workspaceName')  FILTER (WHERE details ->> 'workspaceName' IS NOT NULL))[1] as workspace_name,
+              (ARRAY_AGG(details ->> 'mimeTypes')  FILTER (WHERE details ->> 'mimeTypes' IS NOT NULL))[1] as mime_types
+            FROM ingestion_events
+            WHERE ingest_id LIKE ${if(ingestIdIsPrefix) LikeConditionEscapeUtil.beginsWith(ingestId) else ingestId}
+            AND blob_id NOT IN (SELECT blob_id FROM problem_blobs)
```

Comment:
Is postgres smart enough that if you put this filter here the CTE's are performant? I would have thought that you would need to repeat the `NOT IN (SELECT blob_id FROM problem_blobs)` in blob_extractors and extractor_statuses

Reply:
good call! added to blob_extractors which will also exclude blobs in `problem_blobs` from `extractor_statuses`